src/attr/_make.py

```diff
@@ -446,7 +446,11 @@ def _add_init(cls, frozen):
         sha1.hexdigest()
     )
 
-    script, globs = _attrs_to_script(attrs, frozen)
+    script, globs = _attrs_to_script(
+        attrs,
+        frozen,
+        post_init=hasattr(cls, '__post_init__')
```

Comment:
[No hasattr ever](https://hynek.me/articles/hasattr/) please.  This lends itself to `getattr(cls, "__post_init__", False)`.